src/workers.rs

```diff
@@ -0,0 +1,154 @@
+// Copyright 2018 the Deno authors. All rights reserved. MIT license.
+
+#![allow(dead_code)]
+
+use isolate::Buf;
+use isolate::Isolate;
+use isolate::IsolateState;
+use isolate::WorkerChannels;
+use js_errors::JSError;
+use ops;
+use resources;
+use snapshot;
+use tokio_util;
+
+use futures::sync::mpsc;
+use futures::sync::oneshot;
+use futures::Future;
+use std::sync::Arc;
+use std::thread;
+
+/// Rust interface for WebWorkers.
+pub struct Worker {
+  isolate: Isolate,
```

Comment:
Is Isolate itself just a wrapper around a pointer, or a class with lots of data members?
Asking because if the latter is the case, this inflates every entry in the resource table to something potentially huge.

Reply:
Yes, it's a class with data members - it's not just a pointer. Your comment is valid.

I agree that the resource table should have `Box` elements instead of directly containing structs - but I'd rather do that refactor in a different PR.